### PR TITLE
Move dmp for test compilation in all platforms

### PR DESCRIPTION
**Pull request by JennyChenyj**
- move core to TESTOUTPUT/compile

[skip ci]

Signed-off-by: Jenny Chen <Jenny.Chen@ibm.com>

### CODE REVIEW

File: scripts/resultsSum.pl [hunk 1]
```diff
@@ -135,10 +134,9 @@ sub resultReporter {
 						if (($diagnostic eq 'failure') || ($diagnostic eq 'all')) {
 							$tapString .= $output;
 						}
-						if ($spec =~ /zos/) {
-							my $dmpDir = dirname($resultFile).'/'.$testName;
-							moveTDUMPS($output, $dmpDir);
-						}
+						require "moveDmp.pl";
+						my $dmpDir = dirname($resultFile).'/'.$testName;
+						moveTDUMPS($output, $dmpDir, $spec);
```
Comment (llxia) [inline scripts/resultsSum.pl]: we should keep `if` in resultsSum.pl. During test execution, we only need to take care of zos.

Comment (JennyChenyj) [reply scripts/resultsSum.pl]: Hey Lan! Thanks for your feedback! The change was made. @llxia

**Comment by llxia** [comment]
Tested in https://github.com/eclipse/openj9/pull/7875/

**Comment by llxia** [comment]
We would like to hold this PR until eclipse/openj9#7885 and AdoptOpenJDK/openjdk-tests#1472 are merged

**Comment by llxia** [comment]
could fix the merge conflict and run Grinder_TKG again? Thanks

**Comment by llxia** [comment]
Thanks @JennyChenyj  for testing internally.

**Pull request merged**